www/src/pages/en/usage/trpc.md

```diff
@@ -115,16 +149,122 @@ const userByIdHandler = async (req: NextApiRequest, res: NextApiResponse) => {
 - Instead of creating a response, you can throw errors and return a value or object as you would in any other TypeScript function.
+- Calling the procedure on the frontend doesn't provide and autocompletion or type safety.
+
+## Useful snippets
+
+Here are some snippets that might come in handy.
+
+### Enabling CORS
+
+If you need to consume your API from a different domain, for example in a monorepo that includes a React Native app, you might need to enable CORS:
+
+```ts
+// pages/api/trpc/[trpc].ts
+import type { NextApiRequest, NextApiResponse } from "next";
+import { createNextApiHandler } from "@trpc/server/adapters/next";
+import { appRouter, createContext } from "";
+import cors from "nextjs-cors";
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  // Enable cors
+  await cors(req, res);
+
+  // Create and call the tRPC handler
+  return createNextApiHandler({
+    router: appRouter,
+    createContext: createContext,
+  })(req, res);
+};
+
+export default handler;
+```
+
+### Optimistic updates
+
+Optimistic updates are when we update the UI before the API call has finished. This gives the user a better experience, because they don't have to wait for the API call to finish before the UI reflects the result of their action. However, apps that value data correctness highly should avoid optimistic updates as they are not a "true" representation of backend state. You can read more on the [React Query docs](https://tanstack.com/query/v4/docs/guides/optimistic-updates).
+
+```tsx
+const MyComponent = () => {
+  const listPostQuery = trpc.post.list.useQuery();
+
+  const utils = trpc.useContext();
+  const postCreate = trpc.post.create.useMutation({
+    async onMutate(newPost) {
+      // Cancel outgoing fetches (so they don't overwrite our optimistic update)
+      await utils.post.list.cancel();
+      // Get the data from the queryCache
+      const prevData = utils.post.list.getData();
+      // Optimistically update the data with our new post
+      utils.post.list.setData([...prevData, newPost ]);
+      // Return the previous data so we can revert if something goes wrong
+      return { prevData };
+    },
+    onError(err, newPost, ctx) {
+      // If the mutation fails, use the context-value from onMutate
+      utils.post.list.setData(ctx.prevData);
+    }
+    onSettled() {
+      // Sync with server once mutation has settled
+      utils.post.list.invalidate();
+    },
+  });
+};
+```
+
+### Sample Integration Test
+
+Here is a sample integration test that uses vitest to check that your tRPC router is working as expected, and that the returned data matches the expected output.
+
+```ts
+import { type inferProcedureInput } from "@trpc/server";
+import { createContextInner } from "~/server/router/context";
+import { appRouter, type AppRouter } from "~/server/router/_app";
+import { expect, text } from "vitest";
```

Comment:
```suggestion
import { expect, test } from "vitest";
```

You can have the expect and test as globals just as in Jest, so these imports aren't really necessary.